docs/design/draft/initial-design.md

```diff
@@ -0,0 +1,379 @@
+the only way to identify what to back up.  Will we also allow a list
+of namespaces (including all PVCs in each namespace)? What about
+allowing the use of label selectors? If namespace or label selectors
+are used, we'll also need a Status field to provide the full PV list.
+If these alternate means are supported, do we support all 3 at once
+and take the union of the matches, or do we require that if one is
+specified the others are left blank?
+
+Since other plugins may require additional parameters, we may need a
+map-based `config` element as well. Parameters listed here that relate
+explicitly to PVC configuration can probably be kept as explicit CRD
+fields, though, since any data mover implementation will need to deal
+with them.
+
+Here is an example of a `DataMoverBackup` resource.
+
+```
+apiVersion: oadp.openshift.io/v1alpha1
+kind: DataMoverBackup
+metadata:
+  name: example-backup
+  namespace: oadp-example
+spec:
+  storageLocationRef:
+    name: storage-example
+    namespace: oadp-example
+  pvcs:
+  - name: mysql
+    namespace: mysql-persistent
+  - name: mysql2
+    namespace: mysql-persistent
+  namespaces:
+  - example-project
+  labelSelector:
+    matchLabels:
+      app: mysql
+status:
+  pvcs:
+  - originalPVC:
+      name: mysql
+      namespace: mysql-persistent
+    dataMoverPVC:
+      name: example-backup-q3e4q
+      namespace: mysql-persistent
+    storageClassName: csi-ceph
+    accessModes:
+    - ReadWriteMany
+    storage: 10Gi
+  - originalPVC:
+      name: mysql2
+      namespace: mysql-persistent
+    dataMoverPVC:
+      name: example-backup-6er3q
+      namespace: mysql-persistent
+    storageClassName: csi-ceph
+    accessModes:
+    - ReadWriteMany
+    storage: 10Gi
+
+```
+
+### `DataMoverRestore`
+
+The `DataMoverRestore` is the resource created to initiate a Restore
+operation in the Data Mover controller. The main things that this must
+contain are the `DataMoverBackup` name, a list of PVCs to restore and
+a `StorageLocation` reference.
+
+One open question is what form to provide the Backup reference. We
+can't assume that the `DataMoverBackup` resource will exist in the
+cluster, but we still need an identifying name to use to find the
+backups in the storage location. Whatever identifier is used here must
+be the same one used (possibly as a containing directory in the
+storage location) when backing up the volumes into object storage. We
+could use the Backup name, although this requires Backup names to be
+unique among all clusters using the same storage location. Reusing a
+name in a different cluster could result in data loss. Alternatively,
+we could use the uid from the `DataMoverBackup`, which should
+eliminate that problem. Whatever element is used to identify the
+backup is the one which would be specified here.
+
+
+The main open question here is whether an explicit list of PVCs be
+the only way to identify what to back up.  Will we also allow a list
```

Comment:
should the above sentence be "the only way to identify what to restore" ? (Since we are in restore section).

My 2 cents.

Reply:
Actually this whole paragraph should be removed. Looks like it's a copy-and-paste remnant from the DataMoverBackup section.

Reply:
Thanks for catching it.